models/idea/index.js

```diff
@@ -308,6 +308,52 @@ class Idea extends Model {
     const cursor = await this.db.query(query, params);
     return await cursor.all();
   }
+
+
+  /**
+   * Read trending ideas
+   * @param {integer} offset - pagination offset
+   * @param {integer} limit - pagination limit
+   * @returns {Promise<Idea[]>} - list of found ideas
+   */
+  static async findTrending({ offset, limit }) {
+    const now = + new Date();
+    const oneWeek = 604800000; // 1000 * 60 * 60 * 24 * 7
+    const threeWeeks = 1814400000; // 1000 * 60 * 60 * 24 * 21
+    const threeMonths = 7776000000; // 1000 * 60 * 60 * 24 * 90
+    const weekAgo = now - oneWeek;
+    const threeWeeksAgo = now - threeWeeks;
+    const threeMonthsAgo = now - threeMonths;
+
+    // for each idea we are counting 'rate'
```

Comment:
Arbitrary, but great for now. Then we can see if it ever needs a change based on real world using. :+1: